tools/fork-network/src/cli.rs

```diff
@@ -0,0 +1,285 @@
+        for shard_uid in all_shard_uids {
+            let flat_storage_status = store
+                .get_ser::<FlatStorageStatus>(DBCol::FlatStorageStatus, &shard_uid.to_bytes())?
+                .unwrap();
+            if let FlatStorageStatus::Ready(ready) = &flat_storage_status {
+                let flat_head_for_this_shard = ready.flat_head.hash;
+                if let Some(hash) = flat_head {
+                    if hash != flat_head_for_this_shard {
+                        panic!("Not all shards have the same flat head");
+                    }
+                }
+                flat_head = Some(flat_head_for_this_shard);
+            } else {
+                panic!(
+                    "Flat storage is not ready for shard {}: {:?}",
+                    shard_uid, flat_storage_status
+                );
+            }
+        }
+
+        let flat_head_block = store
+            .get_ser::<near_primitives::block::Block>(
+                DBCol::Block,
+                &flat_head.unwrap().try_to_vec().unwrap(),
+            )?
+            .unwrap();
+
+        let prev_state_roots = flat_head_block
+            .chunks()
+            .iter()
+            .map(|chunk| chunk.prev_state_root())
+            .collect::<Vec<_>>();
+
+        let self_validator = near_config.validator_signer.as_ref().unwrap();
+        let self_account = AccountInfo {
+            account_id: self_validator.validator_id().clone(),
+            amount: 50000000000000000000000000000_u128,
+            public_key: self_validator.public_key().clone(),
+        };
+
+        let mut storage_mutator = StorageMutator::new(
+            epoch_manager.clone(),
+            &*runtime,
+            flat_head_block.header().epoch_id(),
+            *flat_head_block.header().prev_hash(),
+            &prev_state_roots,
+        )?;
+
+        storage_mutator.set_account(
+            self_validator.validator_id().clone(),
+            Account::new(0, self_account.amount, CryptoHash::default(), 0),
+        )?;
+
+        let new_state_roots = storage_mutator.commit()?;
+
+        println!("Creating a new genesis");
+        let epoch_config = epoch_manager.get_epoch_config(&flat_head_block.header().epoch_id())?;
+        let epoch_info = epoch_manager.get_epoch_info(&flat_head_block.header().epoch_id())?;
+        let original_config = near_config.genesis.config.clone();
+
+        let new_config = GenesisConfig {
+            chain_id: original_config.chain_id.clone() + "-fork",
+            genesis_height: flat_head_block.header().height(),
+            genesis_time: flat_head_block.header().timestamp(),
+            epoch_length: self.epoch_length,
+            num_block_producer_seats: epoch_config.num_block_producer_seats,
+            num_block_producer_seats_per_shard: epoch_config.num_block_producer_seats_per_shard,
+            avg_hidden_validator_seats_per_shard: epoch_config.avg_hidden_validator_seats_per_shard,
```

Comment:
For mocknet tests we probably want to make this more configurable.
Better approach would be to make everything more configurable, not just seats.
Maybe we should read GenesisConfig from disk and pass the path to that config as an argument. That way everything in config is configurable. And then we only change validators for the first epoch. 
But it definitely does not matter right now. Just a thought for the future. More for me than for you.

Reply:
Yep, totally. Feel free to play around with this - It's just not clear to me what the best way to feed in input is, but if we can discuss that and agree on something I'm happy to implement that.